api/src/main/java/org/xnio/XnioWorker.java

```diff
@@ -874,6 +877,12 @@ protected final int getWorkerQueueSize() {
         return taskQueue.size();
     }
 
+    public abstract XnioWorkerMXBean getMetrics();
+
+    protected abstract Closeable registerServerMetrics(XnioServerMXBean metrics);
```

Comment:
Make it final on this method, implement the Closeable to add metrics directly to the set which is managed by the Xnio instance.  Basically have the subclasses only implement the interfaces and that's it; put all the control behavior on the superclass.